sdk/storage/AzureStorageBlob/Source/DataStructures/StorageStreamUploader.swift

```diff
@@ -0,0 +1,540 @@
+        client: StorageBlobClient,
+        url: URL,
+        startRange: Int,
+        endRange: Int,
+        options: UploadBlobOptions
+    ) {
+        self.blockId = blockId
+        self.blobName = blob
+        self.containerName = container
+        self.client = client
+        self.options = options
+        self.uploadSource = url
+        self.startRange = startRange
+        self.endRange = endRange
+    }
+
+    // MARK: Public Methods
+
+    /// Begin the upload process.
+    /// - Parameters:
+    ///   - requestId: Unique request ID (GUID) for the operation.
+    ///   - completion: A completion handler that forwards the downloaded data.
+    public func upload(
+        requestId: String? = nil,
+        transactionalContentMd5: Data? = nil,
+        transactionalContentCrc64: Data? = nil,
+        then completion: @escaping HTTPResultHandler<Data>
+    ) {
+        // Construct URL
+        let urlTemplate = "/{container}/{blob}"
+        let pathParams = [
+            "container": containerName,
+            "blob": blobName
+        ]
+        let url = client.url(forTemplate: urlTemplate, withKwargs: pathParams)
+        let chunkSize = endRange - startRange
+        var buffer = Data(capacity: chunkSize)
+        do {
+            guard let fileHandle = try? FileHandle(forReadingFrom: uploadSource) else {
+                throw AzureError.general("Unable to open for reading: \(uploadSource)")
+            }
+            if #available(iOS 13.0, *) {
+                try fileHandle.seek(toOffset: UInt64(startRange))
+            } else {
+                // Fallback on earlier versions
+                fileHandle.seek(toFileOffset: UInt64(startRange))
+            }
+            let tempData = fileHandle.readData(ofLength: chunkSize)
+            buffer.append(tempData)
+        } catch {
+            // TODO: Handle error
+            return
+        }
+
+        // Construct parameters
+        var queryParams = [
+            ("comp", "block"),
+            ("blockid", blockId)
+        ]
+        if let timeout = options.timeout { queryParams.append(("timeout", String(timeout))) }
+
+        // Construct headers
+        var headers = HTTPHeaders([
+            .contentType: "application/octet-stream",
+            .contentLength: String(chunkSize),
+            .apiVersion: client.options.apiVersion
+        ])
+        let leaseAccessConditions = options.leaseAccessConditions
+        let modifiedAccessConditions = options.modifiedAccessConditions
+        let cpk = options.cpk
+
+        if let transactionalContentMd5 = transactionalContentMd5 {
+            headers[.contentMD5] = String(data: transactionalContentMd5, encoding: .utf8)
+        }
+
+        if let transactionalContentCrc64 = transactionalContentCrc64 {
+            headers[.contentCRC64] = String(data: transactionalContentCrc64, encoding: .utf8)
+        }
+
+        if let requestId = requestId { headers["x-ms-client-request-id"] = requestId }
```

Comment:
We should probably use the HTTPHeader and StorageHTTPHeader values here (and below) instead of raw strings.

Reply:
Done.